metadata-ingestion/src/datahub/ingestion/source/usage/bigquery_usage.py

```diff
@@ -256,6 +266,8 @@ class BigQueryUsageConfig(BaseUsageConfig):
     project_id: Optional[str] = None  # deprecated in favor of `projects`
     extra_client_options: dict = {}
     env: str = builder.DEFAULT_ENV
+    table_allow_pattern: Optional[str] = None
```

Comment:
Looks great overall! Could you replace `table_allow_pattern` and `table_deny_pattern` with [AllowDenyPattern](https://github.com/linkedin/datahub/blob/master/metadata-ingestion/src/datahub/configuration/common.py#L52) to make the configs consistent with the general usage of Allow/Deny patterns?
```suggestion
    table_pattern: Optional[AllowDenyPattern] = None
```

Reply:
thanks for the review!

since I am passing this config option as a string to a template, does it still make sense to use that object? can I just do something like `str(table_allow_pattern)` and get the string representation as expected?

Reply:
@rslanka I've updated based on your suggestions, let me know if it works for you! Also went ahead and added a quick test.